packages/dashboard-core-plugins/src/panels/NotebookPanel.tsx

```diff
@@ -681,23 +681,40 @@ class NotebookPanel extends Component<NotebookPanelProps, NotebookPanelState> {
     this.focus();
   }
 
-  handleCopy(): void {
+  async handleCopy(): Promise<void> {
+    const { fileStorage, glEventHub, session } = this.props;
     const { fileMetadata, settings } = this.state;
     assertNotNull(fileMetadata);
-    const content = this.getNotebookValue();
     const { language } = settings;
     const { itemName } = fileMetadata;
-    const copyName = FileUtils.getCopyFileName(itemName);
+    let copyName = FileUtils.getCopyFileName(itemName);
+    // await in loop is fine here, this isn't a parallel task
+    // eslint-disable-next-line no-await-in-loop, @typescript-eslint/strict-boolean-expressions
+    while (await FileUtils.fileExists(fileStorage, copyName)) {
+      copyName = FileUtils.getCopyFileName(copyName);
+    }
```

Comment:
This whole block could be factored into a method in `FileUtils`, e.g.
```
async function getUniqueCopyFileName(fileStorage, originalName) { ... }
```
We do the exact same thing in FileExplorerPanel